### ISSUE TITLE
Unable to fetch image tags for Dockerhub private repo

### ISSUE THREAD

**Issue by sambooo** [weaveworks/flux#896]
As of flux 1.2.0, the cache warmer is failing to fetch image tags from dockerhub private repos.

Example log output:
```
ts=2018-01-11T23:30:04.828130984Z caller=warming.go:154 component=warmer err="requesting tags: errors:\ndenied: requested access to the resource is denied\nunauthorized: authentication required\n"
```

The logs should also make it clear which image we're failing to fetch tags for.

**Comment by rade** [comment]
I think this has also broken the [Weave Cloud Deploy tutorials](https://www.weave.works/docs/tutorials/core/interactive/part-2-deploy-continuous-delivery/)

I am getting
```
ts=2018-01-12T11:22:36.887182409Z caller=warming.go:154 component=warmer err="requesting tags: Get https://<id>-simba01.environments.katacoda.com/v2/root/front-end/tags/list: denied: access forbidden"
```

**Comment by sambooo** [comment]
Fixed by #897
